packages/shared_preferences/shared_preferences/README.md

```diff
@@ -85,6 +85,25 @@ If this is problematic for your use case, you can thumbs up
 interest in APIs that provide direct (asynchronous) access to the underlying
 preference store, and/or subscribe to it for updates.
 
+### Migration and Prefixes
+
+By default, the `SharedPreferences` plugin will only read (and write) preferences
+that begin with the prefix `flutter.`. This is all handled internally by the plugin
+and does not require manually adding this prefix.
+
+If preferences are being migrated from a non-Flutter version of an app,
+they will need to be transformed to include the prefix, or they will not be
+available in the preferences provided by `SharedPreferences`.
+
+Alternatively, `SharedPreferences` can be configured to use any prefix (including 
+no prefix) by adding a call to `setPrefix` before any instances of `SharedPreferences`
+are instantiated. Calling `setPrefix` after an instance of `SharedPreferences` is 
+created will fail.
+
+If you have been using `SharedPreferences` with the default prefix but wish to change
+to a new prefix, you will need to transform your current preferences manually to add 
+the new prefix otherwise the old preferences will be lost.
```

Comment:
I would replace "be lost" with "not be accessible"; they shouldn't actually go anywhere.

I think it's also worth explicitly pointing out that if you set no prefix (which we expect to be the common use case), you can access values previously stored by this plugin by manually adding `flutter.` to the keys.